src/api/pagerenderer.cpp

```diff
@@ -0,0 +1,1080 @@
+  Pta *word_bottom_pts = ptaCreate(0);
+  Pta *word_baseline_pts = ptaCreate(0);
+  Pta *line_baseline_rtl_pts = ptaCreate(0);
+  Pta *line_baseline_ltr_pts = ptaCreate(0);
+  Pta *line_baseline_pts = ptaCreate(0);
+
+  // Replace this with real flags:
+  bool POLYGONFLAG;
+  GetBoolVariable("tessedit_create_page_polygon", &POLYGONFLAG);
+  bool WORDLEVELFLAG;
+  GetBoolVariable("tessedit_create_page_wordlevel", &WORDLEVELFLAG);
+
+  // Use "C" locale (needed for int values larger than 999).
+  page_str.imbue(std::locale::classic());
+  reading_order_str << "\t<Page "
+           <<"imageFilename=\""<< GetInputName();
+  //AppendString(api->GetInputName());
+  reading_order_str << "\" " 
+           << "imageWidth=\"" << rect_width_ << "\" " 
+           << "imageHeight=\"" << rect_height_ << "\" "
+           << "type=\"content\">\n";
+
+  // TODO: Do we need to create a random number here?
+  std::size_t ro_id = std::hash<std::string>{}(GetInputName());
+  reading_order_str << "\t\t<ReadingOrder>\n"
+  << "\t\t\t<OrderedGroup id=\"ro"<< ro_id
+  << "\" caption=\"Regions reading order\">\n";
+
+  ResultIterator *res_it = GetIterator();
+  while (!res_it->Empty(RIL_BLOCK)) {
+    if (res_it->Empty(RIL_WORD)) {
+      res_it->Next(RIL_WORD);
+      continue;
+    }
+
+    int left, top, right, bottom;
+    auto block_type = res_it->BlockType();
+
+    switch (block_type) {
+      case PT_FLOWING_IMAGE:
+      case PT_HEADING_IMAGE:
+      case PT_PULLOUT_IMAGE: {
+        // Handle all kinds of images.
+        page_str << "\t\t<GraphicRegion id=\"r" << rcnt++ << "\">\n";
+        page_str << "\t\t\t";
+        AddBoxToPAGE(res_it, RIL_BLOCK, page_str);
+        page_str << "\t\t</GraphicRegion>\n";
+        res_it->Next(RIL_BLOCK);
+        continue;
+      }
+      case PT_HORZ_LINE:
+      case PT_VERT_LINE:
+        // Handle horizontal and vertical lines.
+        page_str << "\t\t<SeparatorRegion id=\"r" << rcnt++ << "\">\n";
+        page_str << "\t\t\t";
+        AddBoxToPAGE(res_it, RIL_BLOCK, page_str);
+        page_str << "\t\t</SeparatorRegion>\n";
+        res_it->Next(RIL_BLOCK);
+        continue;
+      case PT_NOISE:
+        tprintf("TODO: Please report image which triggers the noise case.\n");
+        ASSERT_HOST(false);
+      default:
+        break;
+    }
+
+    
+    if (res_it->IsAtBeginningOf(RIL_BLOCK)) {
+      // Add Block to reading order
+      reading_order_str << "\t\t\t\t<RegionRefIndexed "
+      << "index=\"" << rcnt << "\" "
+      << "regionRef=\"r" << rcnt << "\"/>\n";
+      
+      float deskew_angle;
+      res_it->Orientation(&orientation_block, &writing_direction_block, &textline_order_block,
+                  &deskew_angle);
+      block_conf = ((res_it-> Confidence(RIL_BLOCK))/100.);
+      page_str << "\t\t<TextRegion id=\"r" << rcnt << "\" "
+      << "custom=\""<< "readingOrder {index:"<< rcnt <<";} "
+      << "readingDirection {"<< WritingDirectionToStr(writing_direction_block)<<";} "
```

Comment:
The `readingDirection` is not required here (and below) for LTR scripts. I'd prefer to write it conditionally only for RTL scripts because it typically reduces the size of the PAGE XML output significantly.

Reply:
Agree.